docs/enhancements/multiple-cpu-platforms.md

```diff
@@ -0,0 +1,144 @@
+---
+title: multiple-cpu-platforms
+authors:
+  - "@danielerez"
+creation-date: 2021-07-29
+last-updated: 2021-07-29
+---
+
+# Multiple CPU platforms support
+
+## Summary
+
+The default host CPU platform supported for installation is x86_64 in Assisted Installer. In order to support additional platforms, the assisted-service should accept and handle selection of other platforms. The focus of this enhancement is specifically arm64, but the introduced infrastructure should facilitate future support of more platforms (e.g. PPC).
+
+## Motivation
+
+Users own machines of numerous variants, hence, support for platforms other than x86_64
+could be highly significant. E.g. creating clusters with arm64 hosts could be crucial
+for some workloads.
+
+### Goals
+
+* The host side images should be built also to arm64.
+* The assisted-service should support the arm64 variant of RHCOS images and OCP release image (for OpenShift >= 4.9).
+* Cluster creation should support specification of CPU platform.
+* Cluster installation flow should be identical from user perspective regardless of the platform.
+
+### Non-Goals
+
+* No need to build assisted-service to arm64. I.e. The assisted-service would remain for x86_64 only.
+* Hybrid clusters - no need to support clusters with hosts of multiple platforms (e.g. x86_64 + arm64).
+* Implementation for kube-api (though design should take into account).
+* Only SNO will be supported in first phase (i.e. no day2/late binding/etc).
+
+## Proposal
```

Comment:
Need to add a host validation that checks that its CPU architecture matches the architecture defined in the cluster property.

Reply:
Isn't it relevant only for late-binding scenarios? I mean, when generating a discovery ISO, we would use only the platform defined in the cluster any way. So no need for additional validation.

Reply:
Yes, my brain always works in late binding mode :)

Reply:
assumed that :) ok, I'll add a comment about it then.